synapse/lib/stormlib/imap.py

```diff
@@ -0,0 +1,376 @@
+                'args': (
+                    {'type': 'str', 'name': 'user',
+                     'desc': 'The username to login with.'},
+                    {'type': 'str', 'name': 'passwd',
+                     'desc': 'The password to login with.'},
+                ),
+                'returns': {
+                    'type': 'list',
+                    'desc': 'An ($ok, $valu) tuple.'
+                },
+            },
+        },
+        {
+            'name': 'search',
+            'desc': '''
+            Search for messages using RFC2060 syntax.
+
+            Examples:
+                Retrieve all messages::
+
+                    ($ok, $uids) = $server.search("ALL")
+
+                Search by FROM and SINCE::
+
+                    ($ok, $uids) = $server.search("FROM", "visi@vertex.link", "SINCE", "01-Oct-2021")
+
+                Search by a subject substring::
+
+                    ($ok, $uids) = $search.search("HEADER", "Subject", "An email subject")
+            ''',
+            'type': {
+                'type': 'function', '_funcname': 'search',
+                'args': (
+                    {'type': 'str', 'name': '*args',
+                     'desc': 'A set of search criteria to use.'},
+                ),
+                'returns': {
+                    'type': 'list',
+                    'desc': 'An ($ok, $valu) tuple, where $valu is a list of UIDs if $ok=True.'
+                },
+            },
+        },
+        {
+            'name': 'select',
+            'desc': 'Select a mailbox to use in subsequent commands.',
+            'type': {
+                'type': 'function', '_funcname': 'select',
+                'args': (
+                    {'type': 'str', 'name': 'mailbox', 'default': 'INBOX',
+                     'desc': 'The mailbox name to select.'},
+                ),
+                'returns': {
+                    'type': 'list',
+                    'desc': 'An ($ok, $valu) tuple.'
+                },
+            },
+        },
+        {
+            'name': 'markSeen',
+            'desc': '''
+            Mark messages as seen by an RFC2060 UID message set.
+
+            The command uses the +FLAGS.SILENT command and applies the \\Seen flag.
+
+            Examples:
+                Mark a single messsage as seen::
+
+                    ($ok, $valu) = $server.markSeen("8182")
+
+                Mark ranges of messages as seen::
+
+                    ($ok, $valu) = $server.markSeen("1:3,6:9")
+            ''',
+            'type': {
+                'type': 'function', '_funcname': 'markSeen',
+                'args': (
+                    {'type': 'str', 'name': 'uid_set',
+                     'desc': 'The UID message set to apply the flag to.'},
+                ),
+                'returns': {
+                    'type': 'list',
+                    'desc': 'An ($ok, $valu) tuple.'
+                },
+            },
+        },
+        {
+            'name': 'markDeleted',
```

Comment:
Could this be simplified to just a `delete()` case that always auto expunges to simplify the use case vs the protocol?

Reply:
updated